File: backend/routers/UserRouter.ts
const express = require("express");
import UserController from "../controllers/UserController";

const userRouter = express.Router();

userRouter.use(express.json());

userRouter.post("/register", UserController.CreateUser);

userRouter.get("/me", UserController.GetMe);

userRouter.get("/me/vendors", UserController.GetMeVendors);

userRouter.put("/:id/password", UserController.UpdatePassword);

userRouter.post("/password/reset/verify", UserController.VerifyPasswordResetToken);

userRouter.post("/password/reset/request", UserController.RequestPasswordReset);

userRouter.post("/login", UserController.Login);

userRouter.post("/logout", UserController.Logout);

userRouter.post("/registration/verify", UserController.VerifyRegistrationToken);

userRouter.post("/registration/verify/new-token", UserController.RequestNewRegistrationToken);

userRouter.put("/delete/:id", UserController.DeleteUser);

userRouter.post("/contact", UserController.Contact);

userRouter.post("/recaptcha/verify", UserController.VerifyRecaptcha);

userRouter.put("/vendors/save", UserController.SaveVendors);

export {userRouter};

File: go-backend/router/user_router.go
package router

import (
	"net/http"

	"handl-server/handlers"

	"github.com/go-chi/chi/v5"
)

// TODO: (HIGH) Implement User routes
func UserRouter() chi.Router {
	r := chi.NewRouter()
	userHandler := &handlers.UserHandler{}

	r.Post("/register", userHandler.CreateUser)

	r.Get("/me", userHandler.GetUserByUserId)

	r.Get("/me/vendors", userHandler.GetSavedVendors)

	r.Put("/{id}/password", userHandler.ResetUserPassword)

	r.Post("/password/reset/verify", func(w http.ResponseWriter, r *http.Request) {

	})

	r.Post("/password/reset/request", userHandler.RequestUserPasswordReset)

	r.Post("/login", userHandler.Login)

	r.Post("/logout", userHandler.Logout)

	r.Post("/registration/verify", func(w http.ResponseWriter, r *http.Request) {

	})

	r.Post("/registration/verify/new-token", func(w http.ResponseWriter, r *http.Request) {

	})

	r.Put("/delete/{id}", userHandler.DeleteUser)

	r.Post("/contact", userHandler.SendSupportMessage)

	r.Post("/recaptcha/verify", func(w http.ResponseWriter, r *http.Request) {

	})

	r.Put("/vendors/save", userHandler.SaveVendors)

	return r
}
